app/models/searchgov_url.rb

```diff
@@ -144,6 +142,27 @@ def validate_document
     raise SearchgovUrlError.new('Noindex per HTML metadata') if document.noindex?
   end
 
+  def check_canonical_url
+    document.canonical_url.present? ? use_canonical_url : index_and_update_status
+  end
+
+  def use_canonical_url
+    canonical_url = document.canonical_url
+    raise SearchgovUrlError, "Record creation forbidden to #{canonical_url}" if redirected_outside_domain?(canonical_url)
+
+    searchgov_url = SearchgovUrl.create(url: canonical_url)
```

Comment:
@lorawoodford Do you think its better to call `fetch` again on new canonical_url to save the new parsed document or just index and update status should be enough?

Reply:
@krbhavith Yes, that makes sense to throw the new/canonical url into the same fetch process as all the rest of the urls.

Also, while we're at it, it would also be great not to have the actual indexing itself buried in the misleadingly named `check_canonical_url` method.  Can we have that method look more like:

```
def check_canonical_url
     return unless document.canonical_url.present?
      
     use_canonical_url
end
```

and then move the `index_and_update_status` back into fetch itself so it is less hidden?  I know this fetch method was a mess before you got here, but we should try to not make it more confusing with these additions.

Reply:
But when I move `index_and_update_status` back to fetch, it is going be called on url when `canonical_url` is created, which we want to avoid that right.

Reply:
I think that depends on what the rewritten/split out `use_canonical_url` ends up looking like.